src/v/cluster/tests/partition_allocator_tests.cc

```diff
@@ -510,3 +510,61 @@ FIXTURE_TEST(change_replication_factor, partition_allocator_fixture) {
     BOOST_CHECK_EQUAL(expected_success.has_value(), true);
     validate_replica_set_diversity(expected_success.value().get_assignments());
 }
+FIXTURE_TEST(rack_aware_assignment_1, partition_allocator_fixture) {
```

Comment:
presumably we could add a ducktape test? we could also probably do it as a follow up PR.

Reply:
I am plus one for the ducktape test

Reply:
I'll add it in a followup PR then

Reply:
https://github.com/redpanda-data/redpanda/issues/4036